### ISSUE TITLE
Binlog-Event is to big - How to skip a binlog event? 

### ISSUE THREAD

**Issue by check24-axel-nitzsche** [krowinski/php-mysql-replication#46]
* *Operating System*: Linux
* *PHP Version*: 7.2.8
* *php-mysql-replication Version*: v5.0.4
* *mysql version*:  Percona 5.7.18

We have a binlog event that is larger than 1 GB and therefore receive the error message `log event entry exceeded max_allowed_packet; Increase max_allowed_packet on master; the first event 'mysql-bin.021114' at 2147481564, the last event read from '/var/log/mysql/mysql-bin.021114' at 123, the last byte read from '/var/log/mysql/mysql-bin.021114' at 2147481583.`.

Unfortunately, the values ​​for `max_allowed_packet` or `slave_max_allowed_packet` can not be set larger than 1 GB.

How can we solve the problem or skip the event?

**Comment by check24-axel-nitzsche** [comment]
We found out the following additional information and corrections:

**_The problem only occurs_**
* Because the binlog file on the database master is 9 GB in size. The individual events are small and the binlog file has become so large because a very large insert / update query has led to many thousands of individual row events.
* If we do not start at the beginning of the binlog file

A work-around could therefore be that always beginning at the beginning of the binlog file and all events are ignored until reaching the relevant position.

**Comment by krowinski** [comment]
Hi, @check24-axel-nitzsche I'm now sure 100% if this will work for you but to skip this event you must use for configuration binLogFileName and binLogPosition. So start from example  bin-001 position 1 then wait until it crash on that big event on position 100 and then restart from position 101. I will try to replicate and add look for permanent solution in free time.

**Comment by c4l3b** [comment]
I ran into this same error today. I found that the max_allowed_packet error can occur for other reasons, including resuming from the wrong spot in a binlog.

My cause today:
I had been setting the `eventsOnly` option, and not including the `ROTATE` event. As the code currently stands, its critical that this event always be included, because otherwise the binlog filename will not be updated here: https://github.com/krowinski/php-mysql-replication/blob/f6033e7e4dbe971802a8879e4f85589a0df3a519/src/MySQLReplication/Event/RotateEvent.php#L26

@krowinski The rotate is a pretty important event to handle internally. Should the user really be able to inadvertently disable the functionality?
